src/main/kotlin/org/rust/ide/inspections/RsVariableMutableInspection.kt

```diff
@@ -0,0 +1,59 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.search.LocalSearchScope
+import com.intellij.psi.search.searches.ReferencesSearch
+import org.rust.ide.annotator.fixes.RemoveMutableFix
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.isMut
+import org.rust.lang.core.psi.ext.parentOfType
+import org.rust.lang.core.psi.ext.selfParameter
+import org.rust.lang.core.resolve.ref.RsMethodCallReferenceImpl
+
+class RsVariableMutableInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "No mutable required"
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
+        object : RsVisitor() {
+            override fun visitPatBinding(o: RsPatBinding) {
+                if (!o.isMut) return
+                val block = o.parentOfType<RsBlock>() ?: o.parentOfType<RsFunction>() ?: return
+                val occurrences = ReferencesSearch.search(o, LocalSearchScope(block))
```

Comment:
`occurrences` declaration is redundant here. This could be:

```rust
ReferenceSearch.search(...)
    .asSequence()
    .map { ... }
    .filter { ... }
    ...
```